Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift

```diff
@@ -48,9 +49,14 @@ extension MainPresenter {
         
         interactor.sessionRequestPublisher
             .receive(on: DispatchQueue.main)
-            .sink { [unowned self] request, context in
-                router.dismiss()
-                router.present(sessionRequest: request, importAccount: importAccount, sessionContext: context)
+            .sink { [unowned self] (request, context) in
+                if let vc = UIApplication.currentWindow.rootViewController?.topController,
+                   vc.restorationIdentifier == SessionRequestModule.restorationIdentifier {
```

Comment:
kind of strange that type cast do not works. Also we can consider isKindOfClass

Reply:
u dont like restorationIdentifier approach?